.devcontainer/devcontainer.json

```diff
@@ -1,37 +1,7 @@
 {
-  // This devcontainer has been set-up to run docker-from-docker scenarios as per
-  // https://github.com/microsoft/vscode-dev-containers/tree/main/containers/docker-from-docker
   "build": {
     "context": "..",
     "dockerfile": "Dockerfile"
   },
-  "runArgs": ["--add-host=host.docker.internal:host-gateway"],
-  "remoteEnv": { "LOCAL_WORKSPACE_FOLDER": "${localWorkspaceFolder}" },
-  "mounts": [
-    "source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind"
-  ],
-  "customizations": {
-    "vscode": {
-      "extensions": [
-        "llvm-vs-code-extensions.vscode-clangd",
-        "marus25.cortex-debug",
-        "matepek.vscode-catch2-test-adapter",
-        "mhutchie.git-graph",
-        "ms-vscode.cmake-tools",
-        "ms-vscode.cpptools",
-        "ms-vsliveshare.vsliveshare",
-        "SonarSource.sonarlint-vscode"
-      ],
-      "settings": {
-        "C_Cpp.autoAddFileAssociations": false,
-        "C_Cpp.intelliSenseEngine": "disabled",
-        "clangd.arguments": [
-          "--query-driver=/**/arm-none-eabi-*"
-        ],
-        "cmake.copyCompileCommands": "${workspaceFolder}/build/compile_commands.json",
-        "cmake.options.statusBarVisibility": "compact",
-        "sonarlint.pathToCompileCommands": "${workspaceFolder}/build/compile_commands.json"
-      }
-    }
-  }
+  "privileged": true
```

Comment:
Running docker container in privileged mode could pose a security risk as it exposes host resources. Do we need this?

Reply:
It depends, I was contemplating on adding it or not. I think I'll remove it here. But if we ever want to debug from within the container this might be necessary.

I'll remove it for now.